pkg/clients/aws/route53.go

```diff
@@ -293,6 +298,149 @@ func NewClient(kubeClient client.Client, secretName, namespace, region string) (
 			MinThrottleDelay: retryerMinThrottleDelaySec * time.Second,
 		},
 	}
+
+	// Check if ClusterDeployment is labelled for STS
+	clusterDeployment := &hivev1.ClusterDeployment{}
+	err := kubeClient.Get(context.TODO(), types.NamespacedName{
+		Name:      clusterDeploymentName,
+		Namespace: namespace,
+	}, clusterDeployment)
+	if err != nil {
+		return nil, err
+	}
+	if stsEnabled, ok := clusterDeployment.Labels["api.openshift.com/sts"]; ok && stsEnabled == "true" {
+		// Get STS jump role from from aws-account-operator ConfigMap
+		cm := &corev1.ConfigMap{}
+		err := kubeClient.Get(context.TODO(), types.NamespacedName{
+			Name:      aaov1alpha1.DefaultConfigMap,
+			Namespace: aaov1alpha1.AccountCrNamespace,
+		}, cm)
+
+		if err != nil {
+			return nil, fmt.Errorf("error getting aws-account-operator configmap to get the sts jump role: %v", err)
+
+		}
+
+		stsAccessARN := cm.Data["sts-jump-role"]
+		if stsAccessARN == "" {
+			return nil, fmt.Errorf("aws-account-operator configmap missing sts-jump-role: %v", aaov1alpha1.ErrInvalidConfigMap)
+		}
+
+		// Get STS Creds
+		secret := &corev1.Secret{}
+		err = kubeClient.Get(context.TODO(),
+			types.NamespacedName{
+				Name:      awsCredsSecretName,
+				Namespace: namespace,
+			},
+			secret)
+
+		if err != nil {
+			return nil, err
+		}
+
+		accessKeyID, ok := secret.Data[awsCredsSecretIDKey]
+		if !ok {
+			return nil, fmt.Errorf("AWS credentials secret %v did not contain key %v",
+				secretName, awsCredsSecretIDKey)
+		}
+
+		secretAccessKey, ok := secret.Data[awsCredsSecretAccessKey]
+		if !ok {
+			return nil, fmt.Errorf("AWS credentials secret %v did not contain key %v",
+				secretName, awsCredsSecretAccessKey)
+		}
+
+		awsConfig.Credentials = credentials.NewStaticCredentials(
+			strings.Trim(string(accessKeyID), "\n"),
+			strings.Trim(string(secretAccessKey), "\n"),
+			"",
+		)
+
+		s, err := session.NewSession(awsConfig)
+		if err != nil {
+			return nil, err
```

Comment:
We might want to provide a more descriptive error here, "unable to setup STS client: %v" or something